telemetry/csharp/ToolkitTelemetryGenerator.Tests/expectedCode.txt

```diff
@@ -0,0 +1,436 @@
+        {
+            if ((value == null))
+            {
+                return;
+            }
+
+            metricDatum.AddMetadata(key, value.ToString());
+        }
+        
+        /// Utility method for generated code to add a metadata to a datum (bool overload)
+        /// Metadata is only added if the value is non-blank
+        private static void AddMetadata(this Amazon.ToolkitTelemetry.Model.MetricDatum metricDatum, string key, bool value)
+        {
+            string valueStr = "false";
+            if (value)
+            {
+                valueStr = "true";
+            }
+
+            metricDatum.AddMetadata(key, valueStr);
+        }
+        
+        /// Utility method for generated code to add a metadata to a datum (double overload)
+        /// Metadata is only added if the value is non-blank
+        private static void AddMetadata(this Amazon.ToolkitTelemetry.Model.MetricDatum metricDatum, string key, double value)
+        {
+            metricDatum.AddMetadata(key, value.ToString(System.Globalization.CultureInfo.InvariantCulture));
+        }
+        
+        /// Utility method for generated code to add a metadata to a datum (int overload)
+        /// Metadata is only added if the value is non-blank
+        private static void AddMetadata(this Amazon.ToolkitTelemetry.Model.MetricDatum metricDatum, string key, int value)
+        {
+            metricDatum.AddMetadata(key, value.ToString(System.Globalization.CultureInfo.InvariantCulture));
+        }
+        
+        /// Records Telemetry Event:
+        /// Emits milliseconds
+        public static void RecordEmitSomeIntMs(this ITelemetryLogger telemetryLogger, EmitSomeIntMs payload)
+        {
+            try
+            {
+                var telemetryEvent = new TelemetryEvent();
+                if (payload.CreatedOn.HasValue)
+                {
+                    telemetryEvent.CreatedOn = payload.CreatedOn.Value;
+                }
+                else
+                {
+                    telemetryEvent.CreatedOn = System.DateTime.Now;
+                }
+                telemetryEvent.Data = new List<Amazon.ToolkitTelemetry.Model.MetricDatum>();
+
+                var datum = new Amazon.ToolkitTelemetry.Model.MetricDatum();
+                datum.MetricName = "emit_someIntMs";
+                datum.Unit = Amazon.ToolkitTelemetry.Unit.Milliseconds;
+                if (payload.Value.HasValue)
+                {
+                    datum.Value = payload.Value.Value;
+                }
+                else
+                {
+                    datum.Value = 1;
+                }
+
+                telemetryEvent.Data.Add(datum);
+                telemetryLogger.Record(telemetryEvent);
+            }
+            catch (System.Exception e)
+            {
+                LOGGER.Error("Error recording telemetry event", e);
```

Comment:
should this bubble back to the caller?

Reply:
Not under the premise that users shouldn't be impacted by telemetry failures. We may want to Debug Assert on this though, so we see it when developing the Toolkit.